pyvelox/substrait.cpp

```diff
@@ -0,0 +1,234 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "substrait.h" // @manual
+#include <google/protobuf/util/json_util.h>
+#include <velox/common/base/Exceptions.h>
+#include <velox/exec/tests/utils/AssertQueryBuilder.h>
+#include <velox/exec/tests/utils/HiveConnectorTestBase.h>
+#include <velox/exec/tests/utils/TempDirectoryPath.h>
+#include <velox/functions/prestosql/aggregates/RegisterAggregateFunctions.h>
+#include <velox/functions/prestosql/registration/RegistrationFunctions.h>
+#include <algorithm>
+#include <filesystem>
+#include <fstream>
+#include <iostream>
+#include <sstream>
+#include <string>
+#include "context.h"
+
+namespace facebook::velox::py {
+
+namespace py = pybind11;
+
+namespace {
+
+bool isFilePath(const std::string& path) {
+  return std::filesystem::exists(path);
+}
+
+static void readFromJSON(
+    const std::string& msg,
+    google::protobuf::Message& protoMsg) {
+  auto status = google::protobuf::util::JsonStringToMessage(msg, &protoMsg);
+  VELOX_CHECK(
+      status.ok(),
+      "Failed to parse Substrait JSON: {} {}",
+      status.code(),
+      status.message());
+}
+
+bool isJSON(std::string content) {
+  // Remove leading whitespaces
+  auto start =
+      std::find_if_not(content.begin(), content.end(), [](unsigned char c) {
+        return std::isspace(c);
+      });
+  content.erase(content.begin(), start);
+
+  // Remove trailing whitespaces
+  auto end =
+      std::find_if_not(content.rbegin(), content.rend(), [](unsigned char c) {
+        return std::isspace(c);
+      });
+  content.erase(end.base(), content.end());
+
+  if (content.empty()) {
+    return false; // If the file is empty, it's not a JSON file
+  }
+
+  // Check if the first character is '{' or '[' and the last character is '}' or
+  // ']'
+  return (
```

Comment:
Is this necessary and sufficient condition for a string being json ?

Reply:
Maybe it would be better to use a JSON validator/identifier standard logic. Should we use an existing API?

Reply:
There are existing api's in folly etc that you can use , rather than write one ourselves..